strapi/strapi
# [Custom Fields] Add custom field forms

### What does it do?

- Adds the user registered forms on custom fields
- Handles save with customField attribute
- Handles validation on the forms
- Adds a new sections to the advanced tab form for other plugins (i.e. i18n) to add custom inputs

### How to test it?
- Run `exmaples/getstarted` with `yarn develop --watch-admin`
- Go to any contentType and click add attribute
- Select the custom tab and click the "Color" custom field, you should see the forms as they are registered in `examples/getstarted/src/plugins/mycustomfields/admin/src/index.js`
- Don't fill the form and hit finish, you should get a validation error on both inputs. The error message on the custom field should be what's registered in examples/getstarted
- Name the field something that already exists on the content type, you should get a validation error
-  Fill out the form and click finish, it should add the attribute to the list
- Click save, it should save to the schema.json:
- 
```
[name]: {
  type: "customField"
  customField: "plugin::mycustomfields.color"
  options: {
     format: 'rgb' | 'hex'
  }
}
```

- Go to any content type where i18n is enabled
- Click the create another field button
- Select the custom tab, choose color, and go to the advanced tab. You should see the Extended settings section with a - checkbox for "Enable localization for this field"
- It should be enabled by default if the content-type has i18n turned on
- Saving a custom field attribute with i18n should save a schem.json like:
```
[name]: {
  "pluginOptions": {
    "i18n": {
      "localized": true | false
    }
  },
  "type": "customField",
  "options": {
     "format": "hex" | "rbga"
  },
  "customField": "plugin::mycustomfields.color"
}
```
